packages/cache-ethereum-storage/src/cache-ethereum-storage.ts

```diff
@@ -0,0 +1,210 @@
+        ipfs: { size: 0 },
+        storageType: StorageTypes.StorageSystemType.ETHEREUM_IPFS,
+        timestamp: Utils.getCurrentTimestampInSecond(),
+      },
+    });
+
+    const resultAppend = await this.ethereumStorage.append(content);
+
+    resultAppend.on('confirmed', (resultAppendConfirmed: StorageTypes.IAppendResult) => {
+      // Remove cache
+      this.data[resultAppend.id] = null;
+
+      result.emit('confirmed', resultAppendConfirmed);
+    });
+
+    // Store in the cache
+    this.data[resultAppend.id] = { content, timestamp: resultAppend.meta.timestamp };
+
+    result.id = resultAppend.id;
+    // TODO Should we add cache?
+    // result.id = `cache:${resultAppend.id}`;
+    result.meta.ipfs = resultAppend.meta.ipfs;
+    result.meta.timestamp = resultAppend.meta.timestamp;
+    result.meta.local = { location: this.locationUrl };
+    result.meta.storageType = StorageTypes.StorageSystemType.LOCAL;
+
+    return result;
+  }
+
+  /**
+   * Read content from the storage
+   * @param Id Id used to retrieve content
+   * @returns Promise resolving content from id
+   */
+  public async read(id: string): Promise<StorageTypes.IEntry> {
+    if (!this.isInitialized) {
+      throw new Error('Ethereum storage must be initialized');
+    }
+
+    if (!id) {
+      throw Error('No id provided');
+    }
+
+    try {
+      // return the data from the ethereum storage
+      return await this.ethereumStorage.read(id);
+    } catch (error) {
+      // if not found, check the cache
+      const cache = this.data[id];
+      if (!cache) {
+        throw Error('No content found from this id');
+      }
+
+      const { content, timestamp } = cache;
+
+      // return the cache information
+      return {
+        content,
+        id,
+        meta: {
+          local: { location: this.locationUrl },
+          storageType: StorageTypes.StorageSystemType.LOCAL,
+          timestamp,
+        },
+      };
+    }
+  }
+
+  /**
+   * Read a list of content from the storage
+   *
+   * @param dataIds A list of dataIds used to retrieve the content
+   * @returns Promise resolving the list of contents
+   */
+  public async readMany(dataIds: string[]): Promise<StorageTypes.IEntry[]> {
+    if (!this.isInitialized) {
+      throw new Error('Ethereum storage must be initialized');
+    }
+
+    const totalCount = dataIds.length;
+    // Concurrently get all the content from the id's in the parameters
+    return Bluebird.map(dataIds, async (dataId, currentIndex) => {
+      const startTime = Date.now();
+      const data = await this.read(dataId);
+      this.logger.debug(
+        `[${currentIndex + 1}/${totalCount}] read ${dataId}. Took ${Date.now() - startTime} ms`,
+        ['read', 'cache'],
+      );
+      return data;
+    });
+  }
+
+  /**
+   * Get all data stored on the storage
+   *
+   * @param boundaries timestamp boundaries for the data retrieval
+   * @returns Promise resolving stored data
+   */
+  public async getData(
```

Comment:
We're only getting the cache here and not the stored data under it.
I guess this function will change a lot depending on how we implement `data`.